internal/services/identitygovernance/access_package_catalog_role_data_source.go

```diff
@@ -0,0 +1,112 @@
+package identitygovernance
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/go-azure-sdk/sdk/odata"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azuread/internal/clients"
+	"github.com/hashicorp/terraform-provider-azuread/internal/tf"
+	"github.com/manicminer/hamilton/msgraph"
+)
+
+func accessPackageCatalogRoleDataSource() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: accessPackageCatalogRoleDataSourceRead,
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"display_name": {
+				Description:  "The display name of the catalog role",
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ExactlyOneOf: []string{"display_name", "object_id"},
+			},
+
+			"template_id": {
+				Description: "The object ID of the template associated with the catalog role",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+
+			"description": {
+				Description: "The description of the catalog role",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+
+			"object_id": {
+				Description:  "The object ID of the catalog role",
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ExactlyOneOf: []string{"display_name", "object_id"},
+			},
```

Comment:
Generally we try to put required attributes first, then optional ones, and computed attributes last. Will update and push this shortly.